File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/interpreter/solution/Nonterminal.m
% Nonterminal.m

classdef (Abstract) Nonterminal < Expression
	methods (Abstract)
		result = getNumberOfUsesOfVariable(theNonterminal, aName)

		result = getValue(theNonterminal, aMap)

		output(theNonterminal)
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/solution/Nonterminal.java
// Nonterminal.java

import java.io.PrintWriter;

public abstract class Nonterminal extends Expression
{
	public abstract int getNumberOfUsesOfVariable(String aName);

	public abstract double getValue(DpMap<String, Double> aMap);

	public abstract void output(PrintWriter aPrintWriter);
}
